src/lib.rs

```diff
@@ -1,40 +1,209 @@
 //! # Restate Rust SDK
 //!
-//! [Restate](https://restate.dev/) is a system for easily building resilient applications using _distributed durable async/await_.
+//! [Restate](https://restate.dev/) is a system for easily building resilient applications.
 //! This crate is the Restate SDK for writing Restate services using Rust.
 //!
+//! ## New to Restate?
+//!
+//! If you are new to Restate, we recommend the following resources:
+//!
+//! - [Learn about the concepts of Restate](https://docs.restate.dev/concepts/durable_building_blocks)
+//! - Use cases:
+//!     - [Workflows](https://docs.restate.dev/use-cases/workflows)
+//!     - [Microservice orchestration](https://docs.restate.dev/use-cases/microservice-orchestration)
+//!     - [Event processing](https://docs.restate.dev/use-cases/event-processing)
+//!     - [Async tasks](https://docs.restate.dev/use-cases/async-tasks)
+//! - [Quickstart](https://docs.restate.dev/get_started/quickstart?sdk=rust)
+//! - [Do the Tour of Restate to try out the APIs](https://docs.restate.dev/get_started/tour/?sdk=rust)
+//!
+//! # SDK Overview
+//!
+//! The Restate Rust SDK lets you implement durable handlers. Handlers can be part of three types of services:
+//!
+//! - [Services](https://docs.restate.dev/concepts/services/#services-1): a collection of durable handlers
+//! - [Virtual Objects](https://docs.restate.dev/concepts/services/#virtual-objects): an object consists of a collection of durable handlers and isolated K/V state. Virtual Objects are useful for modeling stateful entities, where at most one handler can run at a time per object.
+//! - [Workflows](https://docs.restate.dev/concepts/services/#workflows): Workflows have a `run` handler that executes exactly once per workflow instance, and executes a set of steps durably. Workflows can have other handlers that can be called multiple times and interact with the workflow.
+//!
+//! Let's have a look at how to define them.
+//!
+//! ## Services
+//!
+//! [Services](https://docs.restate.dev/concepts/services/#services-1) and their handlers are defined as follows:
+//!
 //! ```rust,no_run
 //! // The prelude contains all the imports you need to get started
 //! use restate_sdk::prelude::*;
 //!
 //! // Define the service using Rust traits
 //! #[restate_sdk::service]
-//! trait Greeter {
-//!     async fn greet(name: String) -> HandlerResult<String>;
+//! trait MyService {
+//!     #[name = "myHandler"]
```

Comment:
Remove the override, it's noisy here. I know it's not consistent with how we name handlers in other languages, but i rather prefer to reduce the noise in this part of the doc.